Src/FluentAssertions/Numeric/NumericAssertions.cs

```diff
@@ -79,9 +79,7 @@ public AndConstraint<TAssertions> Be(T expected, string because = "", params obj
     public AndConstraint<TAssertions> Be(T? expected, string because = "", params object[] becauseArgs)
     {
         Execute.Assertion
-            .ForCondition(
-                (!Subject.HasValue && !expected.HasValue)
-                || (Subject.HasValue && expected.HasValue && Subject.Value.CompareTo(expected.Value) == 0))
+            .ForCondition(expected is T value ? Subject?.CompareTo(value) == 0 : !Subject.HasValue)
```

Comment:
The reduced number of branch hits in this file are due to removal of redundant branches like here.